### ISSUE TITLE
Node 6.8.0 deprecation warnings

### ISSUE THREAD

**Issue by arkady-emelyanov** [oracle/node-oracledb#526]
### install on node 6.8.0

nodejs version: v6.8.0
npm version: 3.10.8
instant client version: 11.2, installed system-wide

command: `npm i` produce lot of deprecation warnings

Example output:

```
../src/njs/src/njsOracle.cpp: In static member function ‘static Nan::NAN_SETTER_RETURN_TYPE Oracledb::SetPoolMin(v8::Local<v8::String>, v8::Local<v8::Value>, Nan::NAN_SETTER_ARGS_TYPE)’:
../src/njs/src/njsUtils.h:362:29: warning: ‘v8::Local<v8::Uint32> v8::Value::ToUint32() const’ is deprecated (declared at /home/vagrant/.node-gyp/6.8.0/include/node/v8.h:8196): Use maybe version [-Wdeprecated-declarations]
     val = v8value->ToUint32()->Value();
...
a lot complains about v8value->ToUint32()->Value();
...

../src/njs/src/njsOracle.cpp:1044:58: warning: ‘v8::Local<v8::Object> v8::Function::NewInstance() const’ is deprecated (declared at /home/vagrant/.node-gyp/6.8.0/include/node/v8.h:3274): Use maybe version [-Wdeprecated-declarations]
                              GetFunction() ->NewInstance();
```

Anyway, module seems working, just lot of deprecations.

**Comment by cjbj** [comment]
This is a NAN issue, isn't it?

**Comment by cjbj** [comment]
Hmm.  Maybe not

**Comment by arkady-emelyanov** [comment]
Same behaviour (lot of deprecations) using Oracle Instant Client 12.1

**Comment by cjbj** [comment]
Node-oracledb probably needs to use NAN wrappers.  I also notice other compile warnings with Node 7 on OS X.  I'll mark it as a bug since some code change will be needed.

**Comment by cjbj** [comment]
There is a fix by @pvenkatraman that will be in the next node-oracledb drop.

**Comment by cjbj** [comment]
Using NAN wrappers has resolved the V8 deprecation warnings.  This is fixed in 1.12.0-dev on GitHub (not pushed to npm). 

There are still a few other, unrelated warnings on OS X; we're aware of them.
